test/jsontest.c

```diff
@@ -0,0 +1,145 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+ */
+
+#include "test.h"
+#include <dps/json.h>
+
+static const char json0[] = "1234";
+static const char json1[] = "-0.5";
+static const char json2[] = "[]";
+static const char json3[] = "[1.0]";
+static const char json4[] = "{}";
+static const char json5[] = "{\"x\":2}";
+static const char json6[] = "true";
+static const char json7[] = "\"hello world\"";
+static const char json8[] = "[1,-1, 2.0, +3.5 ,   -4.0    , -555555  ,6,7,-8,+9]";
+static const char json9[] = "{ \"red\":1, \"green\" : 2, \"blue\": 3 }";
+static const char json10[] =
+"[                      \
+    [1,2,3,4],          \
+    [],                 \
+    [1],                \
+    [true,false],       \
+    \"string\",         \
+    [null,0,\"zilch\"], \
+    [\"a\",\"b\",\"c\"] \
+ ]";
+static const char json11[] =
+"{                            \
+   \"true\" : true,           \
+   \"array1\": [4  , 5],      \
+   \"false\" : false,         \
+   \"array2\":[ 1, 2,3, 4, 5], \
+   \"map1\": {                \
+                \"map2\":{\"1\":1,\"3\":2}\
+             },               \
+   \"null\":null              \
+ }";
+static const char json12[] =
+"[           \
+    { \"$binary\":\"abcdef0123456789\" }, \
+    { \"$binary\" :\"AaBbCcDdEdFf\" }, \
+    { \"$binary\":   \"00\" }, \
+    { \"$binary\":\"\" } \
+ ]";
+
+static const char* tests[] = {
+    json0, json1, json2, json3, json4, json5, json6, json7, json8, json9, json10, json11, json12
+};
+
+static const char bad0[] = "abcd";
+static const char bad1[] = "[ 1,2,3,4,5";
+static const char bad2[] = "{ \"foo\":4 ]";
+static const char bad3[] = "[ 1 2 3 4 ]";
+static const char bad4[] = "{ a:b, c:d }";
+static const char bad5[] = "[[[[ 1, 2 ], 3]}]";
+static const char bad6[] = "ab\ncd";
+static const char bad7[] = "] 1,2 [";
+static const char bad8[] = "[ 1,2,+,= ]";
+
+
+static const char* invalid[] = {
+    bad0, bad1, bad2, bad3, bad4, bad5, bad6, bad7, bad8
+};
+
+static uint8_t cbor1[1024];
+static uint8_t cbor2[1024];
+static char json[4096];
+
+int main(int argc, char** argv)
+{
+    DPS_Status status = DPS_OK;
+    int pretty;
+    int i;
+    // Check with and without formatting
+    for (pretty = 0; pretty <= 1; ++pretty) {
+        for (i = 0; i < (sizeof(tests) / sizeof(tests[0])); ++i) {
+            size_t cbor1Len;
+            size_t cbor2Len;
+
+            // Encode as CBOR
+            status = DPS_JSON2CBOR(tests[i], cbor1, sizeof(cbor1), &cbor1Len);
+            if (status != DPS_OK) {
```

Comment:
replace if (status != DPS_OK) with ASSERT(status == DPS_OK) from test.h

Reply:
Will change to use CHECK macro as used in cbortest.c